src/gleam/result.gleam

```diff
@@ -137,26 +137,36 @@ pub fn flatten(result: Result(Result(a, e), e)) -> Result(a, e) {
 /// ## Examples
 ///
 /// ```gleam
-/// > then(Ok(1), fn(x) { Ok(x + 1) })
+/// > try(Ok(1), fn(x) { Ok(x + 1) })
 /// Ok(2)
 /// ```
 ///
 /// ```gleam
-/// > then(Ok(1), fn(x) { Ok(#("a", x)) })
+/// > try(Ok(1), fn(x) { Ok(#("a", x)) })
 /// Ok(#("a", 1))
 /// ```
 ///
 /// ```gleam
-/// > then(Ok(1), fn(_) { Error("Oh no") })
+/// > try(Ok(1), fn(_) { Error("Oh no") })
 /// Error("Oh no")
 /// ```
 ///
 /// ```gleam
-/// > then(Error(Nil), fn(x) { Ok(x + 1) })
+/// > try(Error(Nil), fn(x) { Ok(x + 1) })
 /// Error(Nil)
 /// ```
 ///
-pub fn then(
+/// `result.try` can be utilized the way
+/// ancient Gleam's `try` used to work:
+///
+/// ```text
+/// // Ancient try
+/// try file = open_file()
+/// // ... can be mimicked like so:
+/// use file <- try(open_file())
+/// ```
+///
```

Comment:
```suggestion
```

Let's not mention the old try, we don't want people getting confused and trying to use that syntax.